.github/workflows/pr.yml

```diff
@@ -23,11 +23,13 @@ jobs:
       - name: Prettier
         run: npm run format
       - name: Check for prettier changes
+        shell: bash
         run: |
-          git diff --exit-code >/dev/null 2>&1
-          EXIT_CODE=$?
+          git diff --exit-code > /dev/null 2>&1 || EXIT_CODE=$?; echo "Exit code: $EXIT_CODE"
           if [ $EXIT_CODE -ne 0 ]; then
             echo "Prettier changes detected. Please run 'npm run format' and commit the changes."
+            echo "Summary of changes: "
+            git diff
```

Comment:
Small tweaks here to make sure the entire snippet runs and logs `git diff` output when there's failure. Was hard for me to debug when it was the `package-lock.json` file causing issues.